plugins/module_utils/mso.py

```diff
@@ -259,14 +393,20 @@ def request(self, path, method=None, data=None, qs=None):
         if qs is not None:
             self.url = self.url + update_qs(qs)
 
+        if method == 'POST' and 'filename' in data['name']:
```

Comment:
I think those changes can be reverted back, right?

Reply:
This is for uploading a backup:
if method == 'POST' and 'filename' in data['name']:
            content_type, data = prepare_multipart(data)
            self.headers['Content-Type'] = content_type

This should work for any file that is to be uploaded within the request function itself.
However, if backup functionality is the only one that uses upload, we can have a separate function for this on mso.py

Actually, I'll create a separate function for this.